File: Python/自己搞定/Easy/125. Valid Palindrome.py
import re
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if (not s or len(s)==1) :
            return True
        s = re.sub("[\W_]","",s).lower()
        middle=int(len(s)/2)
        left = s[:middle]
        if len(s)%2!=0:           
            right = s[middle+1:]
        else:
            right = s[middle:]
        return left==right[::-1]

File: C++/自己搞定/Easy/125. Valid Palindrome.cpp
#include<iostream>
#include <bits/stdc++.h>
#include <map>
using namespace std;

class Solution {
public:
    char ToLower(char word){
        if(word>='A'&& word<='Z'){
            word =word -'A'+'a';
            }
        return word;
    }
    bool checkWord(char word){
        return (word>='a' && word<='z')||(word-'0'>=0 && word-'0'<=9);
    }
    bool isPalindrome(string s) {
        if(s.size()<=1){
            return true;
        }
        int left =0; int right = s.size()-1;bool answer = false;
        while (right>left){
            char leftWord= ToLower(s[left]);
            char rightWord= ToLower(s[right]);    
            if(!checkWord(rightWord)){
                right--;
            }
            else if(!checkWord(leftWord)){
                left++;
            }
            else{         
                if (leftWord!=rightWord){
                    return false;
                }
                else{
                    left++;
                    right--;
                }
            }
        }
        return true;
    }
};